### PR TITLE
(0.87.0) Fix advective forcing for upwind advection schemes

### LINKED ISSUE #2928: Inconsistent advection from "different" velocities
As of now, our tracers are advected by three "different" velocities 
- Velocity
- Background velocity
- Advective forcing

to these, another additional velocity is being added in PR #2802 
All these velocities are applied differently and might have different advection schemes.

This might be a headache in terms of understanding the cause of stability problems. Imagine a situation in which a model is characterized by 3 different velocities that all advect a centered second-order reconstruction of the tracer. If we run into stability problems it might not be immediately clear what the course of action is: is the instability caused by the interaction of the tracer with velocity 1, 2, or 3?

This is not the only issue with this implementation. The implementation is _wrong_ if we use an upwinding reconstruction of the tracer for any of the velocities e.g.:

Imagine we have a Background velocity $U_b = 1$ and an advective forcing $U_a = -1$ and we upwind separately.
The tracer advective flux should be zero everywhere, but, if we upwind:
$$(U_b c) + (U_a c) = c^L - c^R$$

If we want to be able to advect with different velocities consistently we have to ensure that
- the same scheme is used
- we _sum_ the velocities and _not_ apply reconstruction separately

A way to do it could by defining
```
struct AdvectiveU{A, B, C, D}
       u        :: A
       ubkg  :: B
       ufrc   :: C
       ubgc :: D
end

Base.getindex(u::AdvectiveU, i, j, k) = u.u[i, j, k] + u.ubkg[i, j, k] + u.ubcg[i, j, k] + u.ufrc[i, j, k]
```
and passing this velocity to the tracer tendency kernel.

### PR DESCRIPTION
**Pull request by jagoosw**
Closes #2928

This PR fixes advective forcing so that the velocities are correctly added so that upwind and WENO work.

This changes the advective forcing API as it removes the argument `advection_scheme`.

It also removes the scheme and advected tracer properties of the `AdvectiveForcing` struct so scripts using these will no longer work.

### CODE REVIEW

**Comment by glwagner** [comment]
Does this eliminate a feature? What was the application of `advected_field`? Seems like we lose some flexibility if we delete it?

**Comment by jagoosw** [comment]
I think it was only used for selecting the advection operator?

**Comment by glwagner** [comment]
> I think it was only used for selecting the advection operator?

ah indeed

File: src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_tendency_kernel_functions.jl [hunk 1]
```diff
@@ -129,6 +130,9 @@ where `c = C[tracer_index]`.
                         v = SumOfArrays{2}(velocities.v, biogeochemical_velocities.v),
                         w = SumOfArrays{2}(velocities.w, biogeochemical_velocities.w))
 
+        total_velocities = with_advective_forcing(forcing, total_velocities)
+
```
Comment (glwagner) [inline src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_tendency_kernel_functions.jl:134]: one too many lines here
```suggestion
```

**Comment by simone-silvestri** [comment]
We need `advected_field` because it contains the field that is advected by `AdvectiveForcing` (ad also `advective_kernel` which contains the kernel function used to advect)

https://github.com/CliMA/Oceananigans.jl/blob/fa5e280115f619d01a460f012328bd7e6d253b38/src/Forcings/advective_forcing.jl#L65-L66

**Comment by jagoosw** [comment]
> We need `advected_field` because it contains the field that is advected by `AdvectiveForcing` (ad also `advection_kernel_function` which contains the kernel function used to advect)
> 
> https://github.com/CliMA/Oceananigans.jl/blob/fa5e280115f619d01a460f012328bd7e6d253b38/src/Forcings/advective_forcing.jl#L65-L66

Doing it this way we're using the advection kernel function in the normal tendency function instead though:
https://github.com/CliMA/Oceananigans.jl/blob/a33ba461d544a8eda40a26ded20b14f506d72e45/src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_tendency_kernel_functions.jl#L129-L135
so we don't need either any more right?

**Comment by simone-silvestri** [comment]
You're right! I hadn't looked at the whole implementation

**Review by simone-silvestri — APPROVED** [review]
Looks good!

**Comment by simone-silvestri** [comment]
Sure, let's bump minor version

**Comment by jagoosw** [comment]
Merge when tests pass?

**Pull request merged**